python/lsst/analysis/tools/tasks/makeMetricTable.py

```diff
@@ -181,12 +184,14 @@ def run(self, dataIdInfo, metricBundles, skymap):
             corners = self.getTractCorners(skymap, dataIdInfo[0]["tract"])
             metricsDict["corners"] = [corners]
 
-        # Add the metrics from the first bundle to the dict
+        # Add the metrics and units from the first bundle to the dicts
         for name, metrics in metricBundles[0].items():
             for metric in metrics:
                 fullName = f"{name}_{metric.metric_name}"
-                metricValue = metric.quantity
-                metricsDict[fullName] = [metricValue]
+                metricsDict[fullName] = [metric.quantity.value]
+                # "Dimensionless" and percent units not allowed in Tables:
+                if metric.quantity.unit is not u.dimensionless_unscaled and metric.quantity.unit is not u.pct:
+                    metricUnits[fullName] = metric.quantity.unit
```

Comment:
Would it make sense to raise some error here if the unit were not of an allowed type? At the moment, it's just as no-op.

Reply:
Thanks for querying this. `dimensionless_unscaled` just means the metric doesn't have a unit associated with it, so there's nothing to put in the table anyway. In that case an error or warning isn't appropriate - it would effectively be raising an error to say that a metric doesn't have a unit
In the case of percent, I think a warning would be more appropriate; I don't think we want to stop the use of percent units in metrics, we'd just want to warn people that this unit is not being propagated to the table.